File: String/C/replaceChar.c
#include <stdio.h>
#include <string.h>

void replaceCharacter(char *str, char ch, char replace) {
    for (int i = 0; i < strlen(str); i++) {
        if (str[i] == ch) {
            str[i] = replace;
        }
    }
}

int main() {
    char str[] = "This book is very easy";
    char ch = 's';
    char replace = 'p';
    replaceCharacter(str, ch, replace);
    printf("%s\n", str);
    return 0;
}

File: String/Java/replaceChar.java
public class replaceChar {
    public static String replaceCharacter(String str, char ch, char replace) {
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < str.length(); i++) {
            if (str.charAt(i) == ch) {
                result.append(replace);
            } else {
                result.append(str.charAt(i));
            }
        }
        return result.toString();
    }

    public static void main(String[] args) {
        String str = "This book is very easy";
        char ch = 's';
        char replace = 'p';
        System.out.println(replaceCharacter(str, ch, replace));
    }
}